### DISCUSSION TITLE
Action run being trigger multiple times

### DISCUSSION THREAD

**Discussion by lomudi** [community/community#25029] [Actions]
<p>I have added an Action <code>on: release</code> and on each release, it being trigger 3 times in parallel… </p>
<p>What can be the reason for that?</p>
<p>here is my workflow:</p>
<pre><code>on: release
name: Deploy
jobs:
  build:
    name: Deploy
    runs-on: ubuntu-latest
    steps:
    - uses: actions/checkout@master

    - name: Check tag
      uses: actions/bin/filter@master
      with:
        args: tag v*

    - name: Build with NPM
      uses: actions/npm@master
      with:
        args: install
</code></pre>

**Comment by chrispat** [comment, marked as answer]
<p>This is due to GitHub events not really being singular but being loosly a set of events that can happen to a given resource.  If you look at the documentation for the <a href="https://developer.github.com/v3/activity/events/types/#releaseevent" rel="nofollow noopener">release event</a> it will be triggered when a release is published, unpublished, created, edited, deleted and prereleased.</p>
<p>We recently added a feature to the event trigger to let you filter based on these different types of events (doc update coming).  So ifyou want to only trigger when the release is published you could specify that as follows:</p>
<pre><code>on:
  release:
    types: [published]
</code></pre>

**Comment by chiefjester** [comment]
<p>I got bit by this; I’m having two parallel runs instead of 3 though. But to be clear on my end, released is being triggered on both “created” and “published” release event.</p>
<p>I recently found out about the filter option. Before I had to use the Filter Action using the old HCL syntax:</p>
<pre><code class="lang-auto">action "action-filter" {  
uses = "actions/bin/filter@master"  
args = "action published"  
}  
</code></pre>
<p>I’m so glad this is now built-in as mentioned by <a class="mention" href="/chrispat">@chrispat</a> </p>

**Comment by paulineribeyre** [comment]
<p>Hello, the “release” event is triggered by multiple actions: published, unpublished, created, edited, deleted and prereleased. By default, all of these will trigger the workflow. You can specify which type of actions should trigger your workflow, for example:</p>
<pre><code>on:
  release:
    types: [published]
</code></pre>
<p>Source: <a href="https://help.github.com/en/articles/events-that-trigger-workflows#release-event-release" rel="nofollow noopener">the documentation</a>, which I didn’t link directly because there is an error in the example</p>

**Comment by erelsgl** [comment]
<p>I have the same issue but with the “push” event: the action is executed twice in parallel. In the documentation, there are no different activity types for push. Why is my action triggered twice?</p>

**Comment by alvis** [comment]
<p>I hate to say that, but I have the same issue :frowning:</p>

**Comment by mateusfccp** [comment]
<p>This also happens with me. I have three jobs, one for each system, but they run twice each, totalizating 6.</p>

**Comment by lin-ycv** [comment]
I'm still experiencing this issue, even tho i've already specified 

```
 on:
   release:
     type: [published]
```

Anyone can help me on this? it's triggering 3 times for me.

**Reply by Serhiy-dotcom** [reply]
if you still need an answer, then you made a typo you need to write "types" instead of type, so your code should like this:

>  on:
>    release:
>      types: [published]

**Reply by lin-ycv** [reply]
yes, seems like i missed the `s`
